app/Item.php

```diff
@@ -396,6 +397,18 @@ public function getWorkTypesAttribute()
         return $this->makeArray($this->work_type, ', ');
     }
 
+    public function getWorkTypeTreeAttribute()
+    {
+        $workTypes = $this->getWorkTypesAttribute();
+        $tree = $stack = [];
+        foreach ($workTypes as $workType) {
+            $stack[] = $workType;
+            $key = implode(self::TREE_DELIMITER, $stack);
+            $tree[$key] = $workType;
+        }
+        return $tree;
```

Comment:
It took me a while to understand this. It's not really a tree IMHO, but a list of names and their 'paths'

This would've made things a bit more explicit for me:
```suggestion
        return Arr::map($workTypes, function ($workType) use ($stack) {
            $stack[] = $workType;
            return ['name' => $workType, 'path' => implode(self::TREE_DELIMITER, $stack)];
        });
```

Reply:
I had a hard time naming it...